### PR TITLE
Add fusion analytics

### PR DESCRIPTION
**Pull request by wangl5**
### Add fusion analytics
- Add cpprestsdk as a vcpkg dependancy because I couldn't find a nuget http client compatible with vs 2019 and I wanted to avoid adding a library source. I know this isn't ideal, alternative suggestions for http clients are welcome : )
- Add plugin settings file with random tracking ID (SynthesisAddInSettings.json) TODO: Plugin settings manager class
- Updated VS version
- Instead of copying plugin to root directory, the palette and resources folders and the manifest file are copied to the (debug/release) folder. This makes it easier to tell what files are actually part of the built plugin.
- Analytics tracking for all of the same events as the inventor addin

## Build instructions: (for windows command prompt)
  - Install vcpkg: 
    - `git clone https://github.com/Microsoft/vcpkg.git`
    - `cd vcpkg`
    - `bootstrap-vcpkg.bat`
    - `vcpkg integrate install`
  - Install cpprestsdk (this takes a while if you don't have boost installed)
    - `vcpkg install cpprestsdk cpprestsdk:x64-windows`
  - Relaunch visual studio
  - Remove the fusion addin in the `addin` menu in fusion and add the addin in the `synthesis\exporters\FusionRobotExporter\Debug` folder, then stop and start Fusion again

|    DOD    | Done |    Reviewer    | Issue | Notes |
|-----------|------|----------------|-------|-------|
| Code      | ❌   | Kira Corbett/Nicolas Espinoza/Alex Carter|  1056|       |
| Unit Test | N/A  | N/A            |       |       |
| Merged    | ❌  |                |       |       |

### CODE REVIEW

File: exporters/FusionRobotExporter/FusionRobotExporter.vcxproj [hunk 1]
```diff
@@ -70,7 +70,9 @@
       <AdditionalDependencies>core.lib;fusion.lib;cam.lib;kernel32.lib;user32.lib;gdi32.lib;winspool.lib;comdlg32.lib;advapi32.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;odbc32.lib;odbccp32.lib;Rpcrt4.lib;vhacd.lib;%(AdditionalDependencies)</AdditionalDependencies>
     </Link>
     <PostBuildEvent>
-      <Command>xcopy /R /Y "$(SolutionDir)$(Configuration)\$(ProjectName).dll" "$(SolutionDir)"</Command>
+      <Command>xcopy /E /I /Y /S "$(SolutionDir)palette" "$(SolutionDir)$(Configuration)\palette"
+xcopy /E /I /Y /S "$(SolutionDir)Resources" "$(SolutionDir)$(Configuration)\Resources"
+xcopy /Y "$(SolutionDir)$(ProjectName).manifest" "$(SolutionDir)$(Configuration)"</Command>
```
Comment (wangl5) [inline exporters/FusionRobotExporter/FusionRobotExporter.vcxproj]: This is where the palette folder, resources folder, and .manifest file are copied.

File: exporters/FusionRobotExporter/Source/AddIn/Analytics.cpp [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+std::string Analytics::clientId;
+
+void Analytics::setClientID()
+{
+	std::string jsonId = "";
+	try {
+		std::ifstream t("SynthesisAddInSettings.json"); // TODO: settings manager class
+		std::string jsonStr((std::istreambuf_iterator<char>(t)),
+			std::istreambuf_iterator<char>());
```
Comment (wangl5) [inline exporters/FusionRobotExporter/Source/AddIn/Analytics.cpp]: I didn't see a way to save document-independent settings in the fusion api, hence the `SynthesisAddInSettings.json` file. If more plugin-specific settings are needed in the future, a dedicated class should be made to manage the file.

**Comment by wangl5** [comment]
@KiraCorbett Have you installed the `cpprestsdk` library? I put the instructions for installing and using vcpkg in the PR description.

**Comment by KiraCorbett** [comment]
@wangl5 Will retest. Thank you.

**Pull request merged**